aixplain/utils/file_utils.py

```diff
@@ -136,3 +138,69 @@ def upload_data(
             return upload_data(file_name, content_type, content_encoding, nattempts - 1)
         else:
             raise Exception("File Uploading Error: Failure on Uploading to S3.")
+
+def _list_files_and_folders_in_s3_bucket(bucket_name, prefix) -> tuple:
+    # fetch all the files and directory in s3 and return
+    # files, bool if there are dirs inside this s3  
+    try:
+        s3 = boto3.client('s3')
+        response = s3.list_objects_v2(Bucket=bucket_name, Prefix=prefix, Delimiter='/')
+    except Exception as e:
+        raise Exception(f"exception {e}")
+    files = []
+    folders = []
+
+    if 'Contents' in response:
+        for obj in response['Contents']:
+            files.append(obj['Key'])
+
+    if 'CommonPrefixes' in response:
+        return files, bool(response['CommonPrefixes'])
+    else:
+        return files, False
+    
+def _files_to_dict(paths : list, bucket_name : str) -> dict:
+    # convert paths to a dictionary 
```

Comment:
Generate the Docstring of the method as well as the typing of the input variables